custom_components/opnsense/coordinator.py

```
@@ -51,58 +52,44 @@
+
+        for i, cat in enumerate(categories):
+            if not isinstance(results[i], Exception):
+                state[cat.get("state_key")] = results[i]
+            else:
+                _LOGGER.error(
+                    f"Error getting {cat.get('state_key')}. "
+                    f"{results[i].__class__.__qualname__}: {results[i]}"
+                )
+        return state
+
+    async def _get_dhcp_stats(self, leases: list) -> Mapping[str, Any]:
```

Ruff

Missing type annotation for `self` in method

(ANN101)